### PR TITLE
migrations: Call AdoptResources from migrationmaster worker

### PR DESCRIPTION
**Pull request by babbageclunk**
## Description of change
Once the migration is successful we call AdoptResources on the target controller to mark all of the resources used by the model as managed by the target controller. This prevents the source controller from incorrectly cleaning those resources up if it's destroyed.

Which resources are transferred depends on the cloud hosting the model, but it generally includes things like instances, volumes, and security groups.

## QA steps
(Should be done for all clouds, but especially ec2, gce, azure, rackspace, maas, lxd and openstack, since they handle resource tagging and cleanup when a controller is destroyed.)
* bootstrap two controllers, A and B
* add a model on A
* deploy an application the model
* migrate the model to B
* once the migration is complete and the model is running on B, destroy controller A
* the model should continue running on B

## Bug reference
Fixes https://bugs.launchpad.net/juju/+bug/1648063

### CODE REVIEW

File: worker/migrationmaster/worker.go [hunk 1]
```diff
@@ -468,12 +468,29 @@ func (w *Worker) doSUCCESS(status coremigration.MigrationStatus) (coremigration.
...
+		return coremigration.UNKNOWN, errors.Trace(err)
+	}
 	// There's no turning back from SUCCESS - any problems should have
 	// been picked up in VALIDATION. After the minion wait in the
 	// SUCCESS phase, the migration can only proceed to LOGTRANSFER.
 	return coremigration.LOGTRANSFER, nil
 }
 
+func (w *Worker) transferResources(targetInfo coremigration.TargetInfo, modelUUID string) error {
+	w.setInfoStatus("transferring cloud resources to target controller")
```
Comment (axw) [inline worker/migrationmaster/worker.go]: transferring ownership of?

Comment (babbageclunk) [inline worker/migrationmaster/worker.go]: yup, will change this

**Comment by babbageclunk** [comment]
$$merge$$

**Comment by axw** [comment]
$$merge$$

**Comment by babbageclunk** [comment]
Thanks @axw

**Comment by axw** [comment]
$$merge$$

**Pull request merged**